src/buffer/out/textBuffer.cpp

```diff
@@ -1177,37 +1177,36 @@ void TextBuffer::Reset() noexcept
 }
 
 // Arguments:
-// - start: The y-position of the viewport. We'll clear up until here.
-// - height: the number of rows to keep in the buffer.
-void TextBuffer::ClearScrollback(const til::CoordType start, const til::CoordType height)
+// - newFirstRow: The current y-position of the viewport. We'll clear up until here.
+// - rowsToKeep: the number of rows to keep in the buffer.
+void TextBuffer::ClearScrollback(const til::CoordType newFirstRow, const til::CoordType rowsToKeep)
 {
-    // We're already at the top? don't clear ahything. There's no scrollback.
-    if (start <= 0)
+    // We're already at the top? don't clear anything. There's no scrollback.
+    if (newFirstRow <= 0)
     {
         return;
     }
-
     // The new viewport should keep 0 rows? Then just reset everything.
-    if (height <= 0)
+    if (rowsToKeep <= 0)
     {
         _decommit();
         return;
     }
 
-    ClearMarksInRange(til::point{ 0, 0 }, til::point{ _width, start + height });
+    ClearMarksInRange(til::point{ 0, 0 }, til::point{ _width, newFirstRow + rowsToKeep });

```

Comment:
Okay for my next question... wouldn't we then want to only clear up to newFirstRow? The thing I thought was wrong to begin with is quite possibly correct.

Or do we want to clear _even the rows we are keeping_?

(The new names help so so much)